backend/tests/models/user.test.ts

```diff
@@ -29,168 +28,404 @@ describe('User model and validation tests', () => {
+    { desc: 'email extension is not allowed', email: 'testemail@fakesite.xyz' },
+  ];
 
-  it('returns error if email uses capital letters', () => {
-    const testObj = {};
+  it.each(invalidEmails)('returns error if %s', ({ email }) => {
+    const testObj = { email };
     const testOptions = {};
     
-    const { error, value } = User.baseJoi(testObj, testOptions);
+    const { error } = User.baseJoi(testObj, testOptions);
 
-    expect(error).toBeUndefined();
-    expect(value).toBeNull();
+    expect(error).not.toBeNull();
   });
 
-  it('returns error if email extension is not allowed', () => {
-    const testObj = {};
+  const invalidPasswords = [
+    { desc: 'password under 8 characters', password: 'short' },
+    { desc: 'password over 30 characters', password: 'A'.repeat(31) },
+    { desc: 'password uses invalid characters', password: 'Invalid$Password' },
+  ];
+
+  it.each(invalidPasswords)('returns error if %s', ({ password }) => {
+    const testObj = { password };
     const testOptions = {};
     
-    const { error, value } = User.baseJoi(testObj, testOptions);
+    const { error } = User.baseJoi(testObj, testOptions);
 
-    expect(error).toBeUndefined();
-    expect(value).toBeNull();
+    expect(error).not.toBeNull();
   });
 
-  it('returns error if password under 8 characters', () => {
-    const testObj = {};
-    const testOptions = {};
-    
+  it('adds option abortEarly to log all validation errors', () => {
+    const testObj = {
+      email: 'invalid',
+      password: 'short',
+    };
+    const testOptions = { abortEarly: false };
+  
     const { error, value } = User.baseJoi(testObj, testOptions);
-
-    expect(error).toBeUndefined();
-    expect(value).toBeNull();
+  
+    expect(error).not.toBeNull();
+    expect(error?.details.length).toBeGreaterThan(1);
+    expect(value).toEqual(testObj);
   });
 
-  it('returns error if password over 30 characters', () => {
-    const testObj = {};
-    const testOptions = {};
-    
-    const { error, value } = User.baseJoi(testObj, testOptions);
+  const invalidFirstNames = [
+    { desc: 'fname non-alphanumeric', fname: 'John@' },
+    { desc: 'fname less than 1 character', fname: '' },
+    { desc: 'fname over 50 characters', fname: 'A'.repeat(51) },
+  ];
 
-    expect(error).toBeUndefined();
-    expect(value).toBeNull();
-  });
+  const invalidLastNames = [
+    { desc: 'lname non-alphanumeric', lname: 'Doe!' },
+    { desc: 'lname less than 1 character', lname: '' },
+    { desc: 'lname over 50 characters', lname: 'B'.repeat(51) },
+  ];
 
-  it('returns error if password uses characters that are not alphanumeric, numbers, or exclaimation points', () => {
-    const testObj = {};
+  it.each(invalidFirstNames)('returns error if %s', ({ fname }) => {
+    const testObj = { fname };
     const testOptions = {};
     
-    const { error, value } = User.baseJoi(testObj, testOptions);
+    const { error } = User.baseJoi(testObj, testOptions);
```

Comment:
I think there's an issue here and in line 100. baseJoi() doesn't have properties for fname and lname. Did you mean to use registrationJoi() instead?
```suggestion
    const { error } = User.registrationJoi(testObj, testOptions);
```